msal/tests/Test.MSAL.NET.Unit.net45/CoreTests/InstanceTests/B2cAuthorityTests.cs

```diff
@@ -80,26 +85,28 @@ public void NotEnoughPathSegmentsTest()
         [TestCategory("B2CAuthorityTests")]
         public void ValidationEnabledNotSupportedTest()
         {
-            var instance = Authority.CreateAuthority(CoreTestConstants.B2CAuthority, true);
-            Assert.IsNotNull(instance);
-            Assert.AreEqual(instance.AuthorityType, AuthorityType.B2C);
-            try
-            {
-                Task.Run(
-                    async () =>
-                    {
-                        await instance.ResolveEndpointsAsync(
-                            null, 
-                            null,
-                            null, 
-                            new RequestContext(null, new TestLogger(Guid.NewGuid(), null))).ConfigureAwait(false);
-                    }).GetAwaiter().GetResult();
-                Assert.Fail("test should have failed");
-            }
-            catch (Exception exc)
+            using (var httpManager = new MockHttpManager())
             {
-                Assert.IsInstanceOfType(exc, typeof(ArgumentException));
-                Assert.AreEqual(CoreErrorMessages.UnsupportedAuthorityValidation, exc.Message);
+                var instance = Authority.CreateAuthority(CoreTestConstants.B2CAuthority, true);
+                Assert.IsNotNull(instance);
+                Assert.AreEqual(instance.AuthorityType, AuthorityType.B2C);
+                try
+                {
+                    Task.Run(
```

Comment:
can't you do instance.UpdateCannonicalAuthorityAsync(...).GetAwaiter().GetResult() instead? i.e. without the Task.Run ?

Reply:
i can. thanks/

---
In reply to: [230921547](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1383#discussion_r230921547) [](ancestors = 230921547)